bool objects_detected
MapPoint[] map_points